src/pages/Trade/Trade.tsx

```diff
@@ -1,18 +1,105 @@
 import { Flex } from '@chakra-ui/react'
-import { memo } from 'react'
+import { memo, useCallback, useMemo, useRef } from 'react'
+import { FormProvider, useForm } from 'react-hook-form'
 import { useTranslate } from 'react-polyglot'
+import { matchPath, Route, Switch, useHistory, useLocation } from 'react-router-dom'
 
 import { Main } from '@/components/Layout/Main'
 import { SEO } from '@/components/Layout/Seo'
+import { LimitOrder } from '@/components/MultiHopTrade/components/LimitOrder/LimitOrder'
+import { LimitOrderRoutePaths } from '@/components/MultiHopTrade/components/LimitOrder/types'
+import { Claim } from '@/components/MultiHopTrade/components/TradeInput/components/Claim/Claim'
+import { ClaimRoutePaths } from '@/components/MultiHopTrade/components/TradeInput/components/Claim/types'
 import { MultiHopTrade } from '@/components/MultiHopTrade/MultiHopTrade'
+import { TradeInputTab, TradeRoutePaths } from '@/components/MultiHopTrade/types'
 
 const padding = { base: 0, md: 8 }
 
+type MatchParams = {
+  chainId?: string
+  assetSubId?: string
+  sellAssetSubId?: string
+  sellChainId?: string
+  sellAmountCryptoBaseUnit?: string
+}
+
 export const Trade = memo(() => {
   const translate = useTranslate()
+  const location = useLocation()
+  const tradeInputRef = useRef<HTMLDivElement | null>(null)
+  const methods = useForm({ mode: 'onChange' })
+  const history = useHistory()
+
+  // Extract params directly from location.pathname using matchPath instead of useParams()
+  // Somehow, the route below is overriden by /:chainId/:assetSubId/:nftId, so the wrong pattern matching would be used with useParams()
+  // There is probably a nicer way to make this work by removing assetIdPaths from trade routes in RoutesCommon,
+  // and ensure that other consumers are correctly prefixed with their own route, but spent way too many hours on this and this works for now
+  const match = useMemo(
+    () =>
+      matchPath<MatchParams>(location.pathname, {
+        path: `${TradeRoutePaths.Input}/:chainId/:assetSubId/:sellChainId/:sellAssetSubId/:sellAmountCryptoBaseUnit`,
+        exact: true,
+      }),
+    [location.pathname],
+  )
```

Comment:
q: should we extract this path to a constant and consume it in both Trade and MultiHopTrade, or even an util?

Reply:
eyy why not 032e2d0ae8